cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -170,6 +172,14 @@ protected final Ticket getProxiedTicketInstance(final Ticket ticket) {
             return null;
         }
 
+        if (ticket instanceof AccessToken) {
```

Comment:
Read my previous comments. Either we open the `getTicket` method from the `ServiceTicketDelegator` to get the ticket and call the `getAuthentication` method or we have a full OAuth tickets hierarchy.